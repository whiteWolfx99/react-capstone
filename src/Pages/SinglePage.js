import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import GoingBack from '../Components/GoingBack';
import { fetchSingle } from '../redux/SingleSlice';

const SinglePage = () => {
  const { id } = useParams();
  const dispatch = useDispatch();

  const { single, loading, error } = useSelector((state) => state.single);

  useEffect(() => {
    dispatch(fetchSingle(id));
  }, [dispatch, id]);

  return (
    <div className="single_item">
      {loading && <h1 className="fallback">Loading...</h1>}
      {error && <h1 className="fallback">{error}</h1>}
      {single && (
        <div>
          <GoingBack />
          <div className="single">
            <div className="single_item" key={single.id}>
              <img src={single.icon} alt={single.id} />
              <div className="single_name">
                <h2>{single.rank}</h2>
                <h2>{single.name}</h2>
              </div>
              <h3>
                <span> Name:</span>
                {' '}
                {single.name}
              </h3>
              <h3>
                <span> Symbol:</span>
                {' '}
                {single.symbol}
              </h3>
              <h3>
                <span> Rank:</span>
                {' '}
                {single.rank}
              </h3>
              <h3>
                <span> Price:</span>
                {' '}
                {single.price}
                {' '}
                $
              </h3>
              <h3>
                <span> Price in BTC:</span>
                {' '}
                {single.priceBtc}
              </h3>
              <h3 className="single_market">
                <span> Market Cap:</span>
                {' '}
                {single.marketCap}
                {' '}
                $
              </h3>
              <h3>
                <span> Volume:</span>
                {' '}
                {single.volume}
                {' '}
                $
              </h3>
              <h3>
                <span>Available Supply:</span>
                {' '}
                {single.availableSupply}
              </h3>
              <h3>
                <span>Total Supply:</span>
                {' '}
                {single.totalSupply}
              </h3>
              <h3>
                <span>Price Change in hour:</span>
                {' '}
                {single.priceChange1h}
              </h3>
              <h3>
                <span>Price Change in day:</span>
                {' '}
                {single.priceChange1d}
              </h3>
              <h3>
                <span>Price Change in week:</span>
                {' '}
                {single.priceChange1w}
              </h3>
              <div className="url_link">
                <a href={single.websiteUrl} target="_blank" rel="noreferrer">
                  <button type="button" className="single_button">
                    Website
                  </button>
                </a>
                <a href={single.twitterUrl} target="_blank" rel="noreferrer">
                  <button type="button" className="single_button">
                    Twitter
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      )}
    </div>
  );
};
export default SinglePage;
