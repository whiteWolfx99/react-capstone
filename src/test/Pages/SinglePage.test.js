import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import React from 'react';
import store from '../../redux/ConfigureStore';
import SinglePage from '../../Pages/SinglePage';

describe('Detail component', () => {
  it('Rendered Detail component', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <SinglePage />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
