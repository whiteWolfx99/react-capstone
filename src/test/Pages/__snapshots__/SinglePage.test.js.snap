// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail component Rendered Detail component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="single_item"
      >
        <h1
          class="fallback"
        >
          Loading...
        </h1>
        <div>
          <div
            class="going_back"
          >
            <button
              type="button"
            >
              <img
                alt="left arrow"
                src="LeftArrow.svg"
              />
              <h3>
                Go Back
              </h3>
            </button>
          </div>
          <div
            class="single"
          >
            <div
              class="single_item"
            >
              <img />
              <div
                class="single_name"
              >
                <h2 />
                <h2 />
              </div>
              <h3>
                <span>
                   Name:
                </span>
                 
              </h3>
              <h3>
                <span>
                   Symbol:
                </span>
                 
              </h3>
              <h3>
                <span>
                   Rank:
                </span>
                 
              </h3>
              <h3>
                <span>
                   Price:
                </span>
                 
                 
                $
              </h3>
              <h3>
                <span>
                   Price in BTC:
                </span>
                 
              </h3>
              <h3
                class="single_market"
              >
                <span>
                   Market Cap:
                </span>
                 
                 
                $
              </h3>
              <h3>
                <span>
                   Volume:
                </span>
                 
                 
                $
              </h3>
              <h3>
                <span>
                  Available Supply:
                </span>
                 
              </h3>
              <h3>
                <span>
                  Total Supply:
                </span>
                 
              </h3>
              <h3>
                <span>
                  Price Change in hour:
                </span>
                 
              </h3>
              <h3>
                <span>
                  Price Change in day:
                </span>
                 
              </h3>
              <h3>
                <span>
                  Price Change in week:
                </span>
                 
              </h3>
              <div
                class="url_link"
              >
                <a
                  rel="noreferrer"
                  target="_blank"
                >
                  <button
                    class="single_button"
                    type="button"
                  >
                    Website
                  </button>
                </a>
                <a
                  rel="noreferrer"
                  target="_blank"
                >
                  <button
                    class="single_button"
                    type="button"
                  >
                    Twitter
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="single_item"
    >
      <h1
        class="fallback"
      >
        Loading...
      </h1>
      <div>
        <div
          class="going_back"
        >
          <button
            type="button"
          >
            <img
              alt="left arrow"
              src="LeftArrow.svg"
            />
            <h3>
              Go Back
            </h3>
          </button>
        </div>
        <div
          class="single"
        >
          <div
            class="single_item"
          >
            <img />
            <div
              class="single_name"
            >
              <h2 />
              <h2 />
            </div>
            <h3>
              <span>
                 Name:
              </span>
               
            </h3>
            <h3>
              <span>
                 Symbol:
              </span>
               
            </h3>
            <h3>
              <span>
                 Rank:
              </span>
               
            </h3>
            <h3>
              <span>
                 Price:
              </span>
               
               
              $
            </h3>
            <h3>
              <span>
                 Price in BTC:
              </span>
               
            </h3>
            <h3
              class="single_market"
            >
              <span>
                 Market Cap:
              </span>
               
               
              $
            </h3>
            <h3>
              <span>
                 Volume:
              </span>
               
               
              $
            </h3>
            <h3>
              <span>
                Available Supply:
              </span>
               
            </h3>
            <h3>
              <span>
                Total Supply:
              </span>
               
            </h3>
            <h3>
              <span>
                Price Change in hour:
              </span>
               
            </h3>
            <h3>
              <span>
                Price Change in day:
              </span>
               
            </h3>
            <h3>
              <span>
                Price Change in week:
              </span>
               
            </h3>
            <div
              class="url_link"
            >
              <a
                rel="noreferrer"
                target="_blank"
              >
                <button
                  class="single_button"
                  type="button"
                >
                  Website
                </button>
              </a>
              <a
                rel="noreferrer"
                target="_blank"
              >
                <button
                  class="single_button"
                  type="button"
                >
                  Twitter
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
